src/sentry/api/base.py

```diff
@@ -327,7 +327,11 @@ def paginate(
             results = cursor_result.results
 
         response = Response(results)
-        self.add_cursor_headers(request, response, cursor_result)
+
+        # Don't include cursor headers if the client won't be using them
+        if not request.GET.get("no_cursor_headers"):
+            self.add_cursor_headers(request, response, cursor_result)
```

Comment:
Should this be hoisted to the endpoint that is using it? We could skip pagination entirely that way.

Reply:
I considered that, but thought putting it here would mean that if we saw other endpoints bumping up against the header limit they'd be able to use this parameter as well. Though thinking about it a bit more, I'm not sure how often we'll have:
a) An endpoints that's normally paginated
b) But that for a use case we you only want a single page for
c) And because of various reasons is at the header limit

so yeah 🤔 I'll hoist 🧗 this up, and if we notice this happening on other endpoints we can reconsider moving it down again.

Reply:
>  and if we notice this happening on other endpoints we can reconsider moving it down again.

Easier to refactor to an abstraction later over adding a leaky/incorrect abstraction to early.

Reply:
:+1: Another good reason to do it